### PR TITLE
feat: add routes details and domain metadata helper functions

### PR DESCRIPTION
**Pull request by MakMuftic**
## Related Issue Or Context

Closes: #334 #344 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: packages/sdk/src/utils.ts [hunk 1]
```diff
@@ -50,3 +58,71 @@ export async function getTransferStatusData(
...
+ *
+ * @param {Environment} environment - The Sygma environment to use (TESTNET or MAINNET).
+ * @param {number} sourceChainId - The ID of the source chain from which routes are being fetched.
+ * @param {'fungible' | 'gmp' | 'all'} type - The type of the resource for which routes are being fetched. Can be 'fungible', 'gmp', or 'all'.
+ * @returns {Promise<Route[]>} A promise that resolves to an array of Route objects, each representing a route from the source domain to a target domain for a specific resource.
+ * @throws {Error} Throws an error if an invalid environment is specified, if there's a network or server issue during the fetch operation, or if the fetched data cannot be processed correctly.
+ */
+export async function getRoutes(
+  environment: Environment,
+  sourceChainId: number,
```
[the remark this answers is not in the document]
Comment (mpetrunic) [reply packages/sdk/src/utils.ts:105]: nvm found hack

File: packages/sdk/src/types/types.ts [hunk 2]
```diff
@@ -91,3 +91,25 @@ export type TransferStatusResponse = {
...
+  resourceId: string;
+  type: string;
+};
+
+export type DomainMetadata = {
+  url: string; // icon url
+};
+
+export type EnvironmentMetadata = {
+  // domainID -> DomainMetadata
```
Comment (mpetrunic) [inline packages/sdk/src/types/types.ts:107]: does it also work for resourceId?

**Review by mpetrunic — COMMENTED** [review]
You should probably open issue to use this in SDK, as right now if route is unsupported this is error from SDK:
image

😅

**Review by mpetrunic — DISMISSED** [review]

**Comment by MakMuftic** [comment]
## New updates
- removed `axios` as dependency
  - we were using both axios and fetch in codebase, so refactored it to use fetch
- added `isTransferValid` function to BaseAssetTransfer

File: packages/sdk/src/types/types.ts (hunk 2 above)
Comment (MakMuftic) [reply packages/sdk/src/types/types.ts:108]: This will be a separate PR

File: packages/sdk/package.json [hunk 3]
```diff
@@ -84,8 +84,7 @@
     "@polkadot/ui-keyring": "3.4.1",
     "@polkadot/util": "12.2.1",
     "@polkadot/util-crypto": "12.2.1",
-    "axios": "^1.4.0",
-    "cross-fetch": "^3.1.5",
+    "cross-fetch": "^4.0.0",
```
Comment (mpetrunic) [inline packages/sdk/package.json:87]: https://github.com/sygmaprotocol/sygma-sdk/issues/344 😛

**Review by mpetrunic — APPROVED** [review]

**Review by mpetrunic — APPROVED** [review]

**Review by wainola — APPROVED** [review]

**Pull request merged**